File: rulebook-eslint/src/rules/empty-file.ts
import messages from '../messages.js';
import RulebookRule from './rulebook-rule.js';
import type { AST, Rule } from 'eslint';

/** {@link https://hanggrian.github.io/rulebook/rules/#empty-file|See detail} */
class EmptyFileRule extends RulebookRule {
    constructor() {
        super('empty-file', {
            [EmptyFileRule.MSG]: messages.get(EmptyFileRule.MSG),
        });
    }

    override create(context: Rule.RuleContext): Rule.RuleListener {
        return {
            Program(node: AST.Program) {
                // checks for violation
                if (context.sourceCode.getText().trim().length) {
                    return;
                }
                context.report({ node: node, messageId: EmptyFileRule.MSG });
            },
        };
    }

    static MSG: string = 'empty.file';
}

export default new EmptyFileRule();

File: rulebook-cppcheck/rulebook_cppcheck/checkers/empty_file.py
from typing import override

from rulebook_cppcheck.checkers.rulebook_checkers import RulebookFileChecker
from rulebook_cppcheck.messages import _Messages

try:
    from cppcheckdata import Token
except ImportError:
    from cppcheck.Cppcheck.addons.cppcheckdata import Token


class EmptyFileChecker(RulebookFileChecker):
    """See detail: https://hanggrian.github.io/rulebook/rules/#empty-file"""
    ID: str = 'empty-file'
    _MSG: str = 'empty.file'

    @override
    def check_file(self, token: Token, content: str) -> None:
        # checks for violation
        if len(content.strip()) > 0:
            return
        self.report_error(token, _Messages.get(self._MSG))
